tests/testTriggers.go

```diff
@@ -7319,6 +7319,339 @@ func TriggerMetroDR(contexts *[]*scheduler.Context, recordChan *chan *EventRecor
+			taskNamePrefix        = "async-dr-pxrs"
+			allMigrations         []*storkapi.Migration
+			includeVolumesFlag    = true
+			includeResourcesFlag  = true
+			startApplicationsFlag = false
+	)
+
+	Step(fmt.Sprintf("Deploy applications for migration, with frequency: %v", chaosLevel), func() {
+
+			// Write kubeconfig files after reading from the config maps created by torpedo deploy script
+			err := asyncdr.WriteKubeconfigToFiles()
+			if err != nil {
+					log.Errorf("Failed to write kubeconfig: %v", err)
+			}
+
+			err = SetSourceKubeConfig()
+			if err != nil {
+					log.Errorf("Failed to Set source kubeconfig: %v", err)
+			}
+			UpdateOutcome(event, err)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%d-%s", taskNamePrefix, i, time.Now().Format("15h03m05s"))
+					log.Infof("Task name %s\n", taskName)
+					appContexts := ScheduleApplications(taskName)
+					*contexts = append(*contexts, appContexts...)
+					ValidateApplications(*contexts)
+					for _, ctx := range appContexts {
+							// Override default App readiness time out of 5 mins with 10 mins
+							ctx.ReadinessTimeout = appReadinessTimeout
+							namespace := GetAppNamespace(ctx, taskName)
+							migrationNamespaces = append(migrationNamespaces, namespace)
+					}
+					Step("Create cluster pair between source and destination clusters", func() {
+							// Set cluster context to cluster where torpedo is running
+							ScheduleValidateClusterPair(appContexts[0], false, true, defaultClusterPairDir, false)
+					})
+			}
+
+			log.Infof("Migration Namespaces: %v", migrationNamespaces)
+
+	})
+
+	log.InfoD("Start migration")
+
+	for i, currMigNamespace := range migrationNamespaces {
+		migrationName := migrationKey + fmt.Sprintf("%d", i) + time.Now().Format("15h03m05s")
+		currMig, err := asyncdr.CreateMigration(migrationName, currMigNamespace, asyncdr.DefaultClusterPairName, currMigNamespace, &includeVolumesFlag, &includeResourcesFlag, &startApplicationsFlag)
+		if err != nil {
+			UpdateOutcome(event, fmt.Errorf("failed to create migration: %s in namespace %s. Error: [%v]", migrationKey, currMigNamespace, err))
+		} else {
+			allMigrations = append(allMigrations, currMig)
+			Step("Restart Portworx", func() {
+				nodes := node.GetStorageDriverNodes()
+				nodeIndex := rand.Intn(len(nodes))
+				log.Infof("Stop volume driver [%s] on node: [%s]", Inst().V.String(), nodes[nodeIndex].Name)
+				StopVolDriverAndWait([]node.Node{nodes[nodeIndex]})
+				log.Infof("Starting volume driver [%s] on node [%s]", Inst().V.String(), nodes[nodeIndex].Name)
+				StartVolDriverAndWait([]node.Node{nodes[nodeIndex]})
+				log.Infof("Giving a few seconds for volume driver to stabilize")
+				time.Sleep(20 * time.Second)
+			})
+		}
+	}
+
+	// Validate all migrations
+	for _, mig := range allMigrations {
+			err := storkops.Instance().ValidateMigration(mig.Name, mig.Namespace, migrationRetryTimeout, migrationRetryInterval)
+			if err != nil {
```

Comment:
UpdateOutcome has err!=nil check, you can remove if condition here

Reply:
We want custom error, so added err=nil condition.